### ISSUE TITLE
False Positive for E712 with np.where

### ISSUE THREAD

**Issue by franneck94** [astral-sh/ruff#5526]
Hi,

i would expect that this is a false positive for the E712 warning with the np.where function on my pandas data frame.
Because using the "is" keyword here, would be a completly different operation.

![image](https://github.com/astral-sh/ruff/assets/20141069/81052969-7672-406d-a605-3dbcfbbfd13c)

* A minimal code snippet that reproduces the bug.
```python
import numpy as np
import pandas as pd


df_fl_match_id = pd.DataFrame(index=[i for i in range(10)] + [20], columns=["A"])
fl_non_consecutive_idxs_stop = np.where((np.diff(df_fl_match_id.index) != 1) == True)[0]

print(df_fl_match_id)
print(fl_non_consecutive_idxs_stop)
```

* The current Ruff version (`ruff --version`).
ruff 0.0.277

**Comment by charliermarsh** [comment]
Thanks! We have a tracking issue for this in #4560.
